### PR TITLE
setStatusCode method of Response should return $this

### PR DESCRIPTION
**Pull request by kyle-mccarthy**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes

The `setStatusCode` method of `yii\web\Response` should return the Response instance.  This will allow for returning a response in a controller without having issues related to returning void.

```
/**
 * @return \yii\web\Response 
 */
public function actionExample()
{
    $model = new Example();
    if ($model->load(Yii::$app->request->post() && $model->save()) {
        return $this->asJson(['message' => 'Example saved']);
    }
    return $this->asJson(['message' => 'Could not save example', 'errors' => $model->getErrors()])->setResponseCode(400);
}
```

### CODE REVIEW

File: framework/web/Response.php [hunk 1]
```diff
@@ -271,6 +271,7 @@ public function getStatusCode()
      * @param int $value the status code
      * @param string $text the status text. If not set, it will be set automatically based on the status code.
      * @throws InvalidParamException if the status code is invalid.
+     * @return $this
```
Comment (SilverFire) [inline framework/web/Response.php]: `@return $this the response object itself`

File: framework/CHANGELOG.md [hunk 2]
```diff
@@ -41,6 +41,7 @@ Yii Framework 2 Change Log
 - Bug #13379: Fixed `applyFilter` function in `yii.gridView.js` to work correctly when params in `filterUrl` are indexed (SilverFire) 
 - Bug #13670: Fixed alias option from console when it includes `-` or `_` in option name (pana1990)
 - Enh: Added `yii\di\Instance::__set_state()` method to restore object after serialization using `var_export()` function (silvefire)
+- Enh: `setStatusCode` in `\yii\web\Response` returns `$this` (kyle-mccarthy)
```
Comment (SilverFire) [inline framework/CHANGELOG.md]: <code>Enh #13695: \`\yii\web\Response::setStatusCode()\` method now returns the Response object itself (kyle-mccarthy)</code>

**Review by SilverFire — CHANGES_REQUESTED** [review]

**Comment by kyle-mccarthy** [comment]
@SilverFire I made the requested changes and merged with master also.  Let me know if you see anything else!

**Review by SilverFire — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Review by dynasource — APPROVED** [review]

**Comment by 13065176046** [comment]
很不错

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**